spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jdbc/DataSourceAutoConfiguration.java

```diff
@@ -135,6 +140,39 @@ public Object dataSourceMBean(DataSource dataSource) {
 	}
 
 	/**
+	 * Used for avoiding double registration MBean for HikariDataSource.
+	 */
+	@Configuration
+	@ConditionalOnProperty(prefix = "spring.jmx", name = "enabled", havingValue = "true", matchIfMissing = true)
+	@ConditionalOnClass(HikariDataSource.class)
+	@Conditional(DataSourceAutoConfiguration.DataSourceAvailableCondition.class)
```

Comment:
This doesn't work for the example from the description, since bean class may be DataSource.class
In the end, I rewrote to
```
@ConditionalOnBean(DataSource.class)
@ConditionalOnClass(HikariDataSource.class)
```